src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -413,3 +413,70 @@ def do_observation(self, event) -> None:
         self.responses = pd.concat(
             [self.responses, pop_included_mothers] + list(pop_included.values())
         )[self.output_columns]
+
+
+class SocialSecurityObserver(BaseObserver):
+    """Class for observing columns relevant to Social Security registry.
+
+    """
+
+    ADDITIONAL_INPUT_COLUMNS = ["alive", "entrance_time", "exit_time", "ssn"]
+    OUTPUT_COLUMNS = [
+        "first_name_id",
+        "middle_name_id",
+        "last_name_id",
+        "date_of_birth",
+        "event_type",
+        "event_date",
+    ]
+
+    def __repr__(self):
+        return f"SocialSecurityObserver()"
+
+    @property
+    def name(self):
+        return f"social_security_observer"
+
+    @property
+    def output_filename(self):
+        return f"social_security.hdf"
+
+    @property
+    def input_columns(self):
+        return self.DEFAULT_INPUT_COLUMNS + self.ADDITIONAL_INPUT_COLUMNS
+
+    @property
+    def output_columns(self):
+        return self.OUTPUT_COLUMNS
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self.clock = builder.time.clock()
+        self.end_time = dt.datetime(**builder.configuration.time.end)
+
+    def to_observe(self, event: Event) -> bool:
+        """Only observe if this is the final time step of the sim"""
+        return self.clock() <= self.end_time < event.time
+
+    def do_observation(self, event) -> None:
+        pop = self.population_view.get(
+            event.index,
+            query="ssn == True",  # only include simulants with a SSN
+        )
+        df_creation = pop.filter([
+            "first_name_id",
+            "middle_name_id",
+            "last_name_id",
+            "date_of_birth"])
+        df_creation["event_type"] = "creation"
+        df_creation["event_date"] = pop["entrance_time"]
```

Comment:
Note: a large number of people will have their entrance time be the beginning of the simulation, which is not when they would have gotten their SSN. Should we use their birth date instead if it is less than the entrance time? Or maybe we should only record SSN creations that happen after the simulation starts.

Reply:
This was definitely an oversight to not spell out more clearly in the docs, but I think it is heavily implied by the example in the docs that DOB should be used when it is before the simulation start.

Reply:
Thanks for catching this, it looks like the docs are actually pretty clear on this, and it was my oversight, "Everyone who starts in the sim with a SSN will have a creation record made with their date of birth."